### PR TITLE
Update SearchBoxWebPart.ts

### PR DESCRIPTION
**Pull request by kachihro**
Fix for issue #325 (and duplicates).   Handling of search 'inputValue' when an JSON object is determined.

### CODE REVIEW

**Comment by wobba** [comment]
Thank you. I'll let @FranckyC pick this or #330 :) Basically the same fix.

**Comment by kachihro** [comment]
Also - same fix applied to the 'SearchResults' webpart - JSON data chunk for Search QueryText

File: search-parts/src/webparts/searchBox/SearchBoxWebPart.ts [hunk 1]
```diff
@@ -70,15 +70,34 @@ export default class SearchBoxWebPart extends BaseClientSideWebPart<ISearchBoxWe
...
+      }
+      else if (typeof(inputValue === 'object')) {
+        //https://github.com/microsoft-search/pnp-modern-search/issues/325
+        //new issue with search body as object - 2020-06-23
+        const refChunks = this.properties.defaultQueryKeywords.reference.split(':');
+        if (refChunks.length >= 3) {
+          const paramType = refChunks[2];
+  
+          if (paramType === 'fragment') { 
+            this._searchQuery = inputValue["fragment"]
```
Comment (wobba) [inline search-parts/src/webparts/searchBox/SearchBoxWebPart.ts:85]: missing `;`

File: search-parts/src/webparts/searchBox/SearchBoxWebPart.ts [hunk 2]
```diff
@@ -70,15 +70,34 @@ export default class SearchBoxWebPart extends BaseClientSideWebPart<ISearchBoxWe
...
+          }
+          else if (paramType.startsWith('queryParameters')) {
+            const paramChunks = paramType.split('.');
+            const queryTextParam = paramChunks.length === 2 ? paramChunks[1] : 'q';
+            this._searchQuery = inputValue["queryParameters"][queryTextParam]
```
Comment (wobba) [inline search-parts/src/webparts/searchBox/SearchBoxWebPart.ts:90]: missing `; `

Comment (wobba) [inline search-parts/src/webparts/searchBox/SearchBoxWebPart.ts:90]: Need to check if `q` exists when doing fallback.

File: search-parts/src/webparts/searchBox/SearchBoxWebPart.ts (hunk 1 above)
Comment (wobba) [reply search-parts/src/webparts/searchBox/SearchBoxWebPart.ts:85]: Should decode it `decodeURIComponent(inputValue["fragment"])`

File: search-parts/src/webparts/searchResults/SearchResultsWebPart.ts [hunk 3]
```diff
@@ -205,6 +205,25 @@ export default class SearchResultsWebPart extends BaseClientSideWebPart<ISearchR
...
+        if (typeof(queryKeywords) == "object")
+        {
+            //https://github.com/microsoft-search/pnp-modern-search/issues/325
+            //new issue with search body as object - 2020-06-23
+            const refChunks = this.properties.queryKeywords.reference.split(':');
+            if (refChunks.length >= 3) {
+                const paramType = refChunks[2];
+        
+                if (paramType === 'fragment') { 
+                    queryKeywords = queryDataSourceValue["fragment"]
```
Comment (wobba) [inline search-parts/src/webparts/searchResults/SearchResultsWebPart.ts:217]: Same as comment for search box

**Comment by wobba** [comment]
@kachihro Would be nice if you can address the comments, and also test the fix for when you consume search query from the header search box.

**Comment by wobba** [comment]
@kachihro I added the extra checks to your code and pushed to dev.

**Comment by kachihro** [comment]
> @kachihro I added the extra checks to your code and pushed to dev.

Thanks - sorry, I hadn't gotten to it today, was just sitting down to check now !      👍

**Pull request closed without merging**